src/tokenizer.rs

```diff
@@ -1224,27 +1224,22 @@ pub fn parse_raw_string_literal(
     match stream.get_next() {
         Some('"') => pos.advance(),
         Some(c) => return Err((LERR::UnexpectedInput(c.to_string()), start)),
-        None => {
-            return Err((LERR::UnterminatedString, start));
-        }
+        None => return Err((LERR::UnterminatedString, start))
     }
 
-    let mut seen_hashes: Option<u8> = None;
     // Match everything until the same number of '#'s are seen, prepended by a '"'
+
+    // Counts the number of '#' characters seen after a quotation mark.
+    // Becomes Some(0) after a quote is seen, but resets to None if a hash doesn't follow.
+    let mut seen_hashes: Option<u8> = None;
     let mut result = SmartString::new_const();
 
+
     loop {
         let next_char = match stream.get_next() {
-            Some(ch) => {
-                pos.advance();
-                ch
-            }
-            None => {
-                pos.advance();
-                return Err((LERR::UnterminatedString, start));
-            }
+            Some(ch) => ch,
+            None => return Err((LERR::UnterminatedString, start))
         };
-        pos.advance();
```

Comment:
Turns out all of the `pos.advances()` above this one were erroneous anyways! The new test code checks this.